File: 2596. Check Knight Tour Configuration/Solution.py
class Solution:
    def checkValidGrid(self, grid: List[List[int]]) -> bool:
        movements = [[2, 1], [2, -1], [-2, 1], [-2, -1], [1, 2], [1, -2], [-1, 2], [-1, -2]]
        q = [[0, 0]]
        depth, n = 0, len(grid)
        while q:
            l = len(q)
            for i in range(l):
                x, y = q.pop(0)
                if grid[x][y] != depth: continue
                if grid[x][y] == n * n - 1: return True
                for a, b in movements:
                    nx, ny = x + a, y + b
                    if nx < 0 or ny < 0 or nx >= n or ny >= n: continue
                    q.append([nx, ny])
            depth += 1
        return False

File: 2596. Check Knight Tour Configuration/Solution.java
class Solution {
    public boolean checkValidGrid(int[][] grid) {
        int[][] movements = new int[][] { { 2, 1 }, { 2, -1 }, { -2, 1 }, { -2, -1 }, { 1, 2 }, { 1, -2 }, { -1, 2 }, { -1, -2 } };
        Queue<int[]> q = new LinkedList<>();
        q.offer(new int[] { 0, 0 });
        int depth = 0, n = grid.length;
        while (!q.isEmpty()) {
            int l = q.size();
            for (int i = 0; i < l; i++) {
                int[] curr = q.poll();
                if (grid[curr[0]][curr[1]] != depth) continue;
                if (depth == n * n - 1) return true;
                for (int[] m : movements) {
                    int nx = curr[0] + m[0], ny = curr[1] + m[1];
                    if (nx < 0 || ny < 0 || nx >= n || ny >= n) continue;
                    q.offer(new int[] { nx, ny });
                }
            }
            depth++;
        }
        return false;
    }
}